services/imageproxy/local.go

```diff
@@ -66,6 +69,26 @@ func makeLocalBackend(proxy *ImageProxy) *LocalBackend {
 	}
 }
 
+type contentTypeRecorder struct {
+	http.ResponseWriter
+	filename string
+}
+
+func (rec *contentTypeRecorder) WriteHeader(code int) {
+	defer rec.ResponseWriter.WriteHeader(code)
+
+	hdr := rec.ResponseWriter.Header()
+	contentType := hdr.Get("Content-Type")
+	mediaType, _, err := mime.ParseMediaType(contentType)
+	if err != nil {
+		mlog.Error("error in decoding media type", mlog.Err(err))
+		return
+	}
+	if mediaType == "image/svg+xml" {
```

Comment:
The error case should either set `Content-Disposition` or somehow bubble up the error to the client: We don't know what the MIME type is or how the client will interpret it, so it could still be SVG.
```suggestion
	if err != nil || mediaType == "image/svg+xml" {
```

Reply:
It might need more work to bubble up the error to the client since it's a middleware. Do you not want to log the error at all ?

Reply:
Logging it doesn't seem strictly necessary since this is an error caused by malformed input from the external image server -- and it's not causing anything to actually fail. But I'm not opposed to logging the error either. Bubbling up probably doesn't make sense, I agree.

Reply:
Alright, I'll make the changes.